src/Microsoft.TemplateEngine.Abstractions/ITemplateEngineHost.cs

```diff
@@ -17,29 +20,46 @@ public interface ITemplateEngineHost
 
         IReadOnlyList<string> FallbackHostTemplateConfigNames { get; }
 
+        /// <summary>
+        /// Gets default logger for given template engine host.
+        /// </summary>
+        ILogger Logger { get; }
+
+        /// <summary>
+        /// Gets logger factory for given template engine host.
+        /// </summary>
+        ILoggerFactory LoggerFactory { get; }
+
         string Version { get; }
 
+        [Obsolete("Use " + nameof(Logger) + " instead.")]
         void LogTiming(string label, TimeSpan duration, int depth);
 
+        [Obsolete("Use " + nameof(Logger) + " instead.")]
         void LogMessage(string message);
 
+        [Obsolete("Use " + nameof(Logger) + " instead.")]
         void OnCriticalError(string code, string message, string currentFile, long currentPosition);
 
+        [Obsolete("Use " + nameof(Logger) + " instead.")]
         bool OnNonCriticalError(string code, string message, string currentFile, long currentPosition);
 
-        // return of true means a new value was provided
+        [Obsolete("remove candidate")]
```

Comment:
I suggest to remove this and 2 other methods below, thoughts?
Also suggest to replace OnPotentiallyDestructiveChangesDetected with error / exception instead of event.

Reply:
https://github.com/dotnet/templating/issues/3146